### PR TITLE
Dag for smd 311 csrs

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/16858

## Associated repo

https://github.com/cityofaustin/atd-knack-services/pull/141

## Testing

https://datahub.austintexas.gov/Transportation-and-Mobility/Signs-and-Markings-Knack-311-Service-Requests/ji6n-8jff/about_data

The knack services PR has been merged, so you can run locally without having to make a special atd-knack-services docker container. 

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Add note to 1PW secrets moved to API vault and check for duplicates

### CODE REVIEW

File: dags/atd_knack_smd_311_csr.py
Comment (chiaberry) [inline dags/atd_knack_smd_311_csr.py:1]: this is the new dag, the rest was me adding the `docker_conn_id` to other signs and markings dags

**Review by mddilley — APPROVED** [review]
There is something wrong with my local 1PW setup so I couldn't get this up and running, but the DAG code looks good to me! Thanks for adding the Docker param to all these DAGs! 🙏 🚢

**Review by Charlie-Henry — APPROVED** [review]
Looks good, didn't test 🚢

File: dags/atd_knack_smd_311_csr.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        "opfield": "socrata.appToken",
+    },
+}
+
+
+with DAG(
+    dag_id="atd_knack_signs_markings_311_csr",
+    description="Publish signs markings 311 CSRs to Postgres, Socrata",
+    default_args=DEFAULT_ARGS,
+    schedule_interval="20 19 * * *" if DEPLOYMENT_ENVIRONMENT == "production" else None,
```
Comment (frankhereford) [inline dags/atd_knack_smd_311_csr.py:60]: Just curious - is there something special about 7:20 PM?

**Review by frankhereford — APPROVED** [review]
I ran this locally, and it worked great. Thanks for adding this DAG and adding the connection ID to the other slew of them. 🙏 

t1:
```
[2024-04-26, 08:11:35 CDT] {docker.py:324} INFO - Starting docker container from image atddocker/atd-knack-services:production
[2024-04-26, 08:11:36 CDT] {docker.py:391} INFO - /app/./atd-knack-services/services/records_to_postgrest.py.INFO: Namespace(app_name='signs-markings', container='view_3992', date='2024-04-26T13:10:29.518119+00:00')
[2024-04-26, 08:11:36 CDT] {docker.py:391} INFO - /app/./atd-knack-services/services/records_to_postgrest.py.INFO: Downloading records from Knack...
[2024-04-26, 08:12:00 CDT] {docker.py:391} INFO - /app/./atd-knack-services/services/records_to_postgrest.py.INFO: 25 to process.
[2024-04-26, 08:12:00 CDT] {docker.py:391} INFO - /app/./atd-knack-services/services/records_to_postgrest.py.INFO: Records uploaded: 25
[2024-04-26, 08:12:00 CDT] {taskinstance.py:1350} INFO - Marking task as SUCCESS. dag_id=atd_knack_signs_markings_311_csr, task_id=smd_311_csrs_to_postgrest, execution_date=20240426T131028, start_date=20240426T131031, end_date=20240426T131200
[2024-04-26, 08:12:00 CDT] {local_task_job_runner.py:225} INFO - Task exited with return code 
```
t2:
```
[2024-04-26, 08:12:01 CDT] {docker.py:324} INFO - Starting docker container from image atddocker/atd-knack-services:production
[2024-04-26, 08:12:02 CDT] {docker.py:391} INFO - /app/./atd-knack-services/services/records_to_socrata.py.INFO: Namespace(app_name='signs-markings', container='view_3992', date='2024-04-26T13:10:29.518119+00:00')
[2024-04-26, 08:12:03 CDT] {docker.py:391} INFO - /usr/local/lib/python3.11/site-packages/knackpy/app.py:60: UserWarning: No API key has been supplied. Only public views will be accessible.
  warnings.warn(
[2024-04-26, 08:12:03 CDT] {docker.py:391} INFO - /app/./atd-knack-services/services/records_to_socrata.py.INFO: Downloading records from app 5d13ae5b438091000ac0197d, container view_3992.
[2024-04-26, 08:12:03 CDT] {docker.py:391} INFO - /app/./atd-knack-services/services/records_to_socrata.py.INFO: 25 records to process
[2024-04-26, 08:12:04 CDT] {docker.py:391} INFO - /app/./atd-knack-services/services/records_to_socrata.py.INFO: Record field names not in Socrata: ['id']
[2024-04-26, 08:12:04 CDT] {docker.py:391} INFO - /app/./atd-knack-services/services/records_to_socrata.py.INFO: 25 records processed.
[2024-04-26, 08:12:04 CDT] {taskinstance.py:1350} INFO - Marking task as SUCCESS. dag_id=atd_knack_signs_markings_311_csr, task_id=smd_311_csrs_to_socrata, execution_date=20240426T131028, start_date=20240426T131201, end_date=20240426T131204
[2024-04-26, 08:12:04 CDT] {local_task_job_runner.py:225} INFO - Task exited with return code 0
```

File: dags/atd_knack_smd_311_csr.py (hunk 1 above)
Comment (chiaberry) [reply dags/atd_knack_smd_311_csr.py:60]: nothing special, i just picked a time not during normal working hours. i will check with Christina if this time works

Comment (frankhereford) [reply dags/atd_knack_smd_311_csr.py:60]: Totally all good, please don't check on my account. I was just curious if it lined up with some outside event or something like that. Thanks again for getting this set up in airflow!

**Pull request merged**